packages/interactivity/router/index.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * WordPress dependencies
+ */
+import {
```

Comment:
Can we avoid files outside "src" and just move this router to its own `@wordpress/interactivity-router` package instead?

Reply:
Also, can I ask why do we want to split this into its own module rather than having it part of the interactivity module?

Reply:
Looking at the description:

<img width="550" alt="Screenshot 2024-01-24 at 10 18 25" src="https://github.com/WordPress/gutenberg/assets/699132/165a76ea-712b-4a49-afd3-3af72caae6ed">

There aren't many savings for the runtime bundle, something around 0.4 kb,

Reply:
Sure! Working on it.

Reply:
@DAreRodz Creating modules and packages comes with extra maintenance cost, so we need to make sure that there are good gains to be made.

0.4kb might not be worth it, no? Why do you want to separate this module?

Reply:
Done in https://github.com/WordPress/gutenberg/pull/57924/commits/7b9e112e020037609927449b79025205abcaf62c.

Reply:
> 0.4kb might not be worth it, no? Why do you want to separate this module?

I know it could not be much of a saving here, although we expect that module to increase a bit in complexity fairly soon. For example, we want to absorb logic to handle accessibility and visuals like the top loading bar (logic that is currently inside the Query block).

Also, our idea was to start testing modules imported dynamically to see how that approach would be implemented.

Does it make sense? @gziolo @youknowriad

Reply:
Yes, the router today is still very basic, but it will start increasing in size and complexity once we start adding more features. By decoupling it from the main runtime, we ensure that we can add those features without affecting the performance of WordPress sites that don't use client-side navigation. The router is also a part that is not required when the page is loaded, so even sites that use client-side navigation can benefit from the code-splitting. 

This is more personal, but I also consider that if devs learning the Interactivity API are exposed to these code-splitting patterns because they are used by the framework itself, it will be more likely that they learn and apply them in their own code.

Finally, we also want to start doing code-splitting ourselves to test that code-splitting works fine in both the Interactivity API and the new Script Modules API before this technique is exposed to third-party plugins.

I don't have strong feelings about `@wordpress/interactivity/router` vs `@wordpress/interactivity-router` though. Whatever you feel is better/simpler works for me 🙂